### ISSUE TITLE
Back/forward cache becomes broken if some script is active on the page

### ISSUE THREAD

**Issue by Inversion-des** [Tampermonkey/tampermonkey#1491]
Shown reason — **OptInUnloadHeaderNotPresent** (The page has an unload handler in the main frame).
If Tampermonkey is active without any scripts — all is ok.

## Expected Behavior
Tampermonkey should not affect on Back/forward cache (except if you explicitly add .onunload handler to the page).

## Actual Behavior
Back/forward cache is just broken. You can see and check it in DevTools:

![image](https://user-images.githubusercontent.com/1228795/164621505-3a63236a-d986-49a2-b26c-fe2cd9346fe5.png)

## Specifications
- Chrome: 100.0.4896.127
- TM: 4.16
- OS: Win10 pro

## Script
(simple script is enough)

```js
// ==UserScript==
// @name         Remove some params from URL
// @version      0.1
// @include      *
// @grant        none
// ==/UserScript==

(function() {
    // for link opened from Facebook
    var param = 'fbclid';
    if (location.search.indexOf(param + '=') !== -1) {
        var replace = '';
        try {
            var url = new URL(location);
            url.searchParams.delete(param);
            replace = url.href;
        } catch (ex) {
            var regExp = new RegExp('[?&]' + param + '=.*$');
            replace = location.search.replace(regExp, '');
            replace = location.pathname + replace + location.hash;
        }
        history.replaceState(null, '', replace);
    }
    else {
        if (location.search.match(/\?(tl_inbound|utm_campaign|utm_source)/)) {
            history.replaceState(null, '', location.pathname)
        }
    }
})();
```

**Comment by derjanb** [comment]
Looks like some extension in general block back/forward cache.

![image](https://user-images.githubusercontent.com/767504/172915938-1387f39b-68fd-40e1-846c-293134b32065.png)

**Comment by Inversion-des** [comment]
You should try a different Chrome instance (profile). This one looks broken to me…

**Comment by derjanb** [comment]
Should be fixed at 5.3.6204 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx))

Please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by Inversion-des** [comment]
Works fine for me in Beta v5.3.6205. 
But what is strange: when I tried today to reproduce the problem with my current v5.1.1 — I could not.
Anyway, thank you!
